dockerfile-image-update/src/main/java/com/salesforce/dockerfileimageupdate/search/GitHubImageSearchTermList.java

```diff
@@ -0,0 +1,145 @@
+     * @param state processing state
+     * @param imagePart the current image segment
+     */
+    private static void processIntermediateUrlSegment(ProcessingState state, String imagePart) {
+        if (imagePart.contains("-")) {
+            state.finalizeCurrentTerm();
+        } else {
+            state.addToCurrentTerm("/");
+        }
+        state.addToCurrentTerm(imagePart);
+    }
+
+    /**
+     * The final URL segment will conclude the current search term if what is currently in the buffer has dashes.
+     * Regardless, the final search term should have a slash separator whether it's part of the current search
+     * term or at the beginning of a new term.
+     *
+     * @param state processing state
+     * @param finalImageSegment the final URL segment
+     */
+    private static void processFinalUrlSegment(ProcessingState state, String finalImageSegment) {
+        String leftoverTerm = state.getCurrentTerm();
+        if (leftoverTerm.contains("-")) {
+            state.finalizeCurrentTerm();
+        }
+        state.addToCurrentTerm("/");
+        state.addToCurrentTerm(finalImageSegment);
+    }
+
+    /**
+     * Domains with dashes do not function well with GitHub code search
+     * https://docs.github.com/en/github/searching-for-information-on-github/searching-code#considerations-for-code-search
+     *
+     * Unfortunately, code search is not implemented in any other way such as GraphQL
+     *
+     * @param domain the domain part of the image (which may or may not be the full registry name)
+     * @return processing state for the search terms
+     */
+    static ProcessingState processDomainPartOfImage(String domain) {
+        ProcessingState state = new ProcessingState();
+        if (domain == null || domain.trim().isEmpty()) {
+            return state;
+        } else {
+            state.addToCurrentTerm("FROM ");
+            if (domain.contains("-")) {
+                processDashedDomainParts(state, domain);
+            } else {
+                state.addToCurrentTerm(domain);
+            }
+        }
+        return state;
+    }
+
+    /**
+     * Dashed domain parts should simply by split by dashes and we'll leave the final domain part in the current search
+     * term buffer
+     *
+     * @param state processing state
+     * @param domain the full dashed domain
+     */
+    private static void processDashedDomainParts(ProcessingState state, String domain) {
+        String[] domainParts = domain.split("-");
+        for (int i = 0; i < domainParts.length - 1; i++) {
+            state.addToCurrentTerm(domainParts[i]);
+            state.finalizeCurrentTerm();
+        }
+        state.addToCurrentTerm(domainParts[domainParts.length - 1]);
+    }
+
+    /**
+     * Helper class to store processing state of a search term or terms
+     */
+    static class ProcessingState {
+        final List<String> terms = new ArrayList<>();
+        private StringBuffer termBuffer = new StringBuffer();
```

Comment:
Shouldn't this be `final`? It might be safer do use https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/StringBuffer.html#delete(int,int) below?

Reply:
Yeah, fair point - inbound change coming to `final` and `setLength(0)`

Reply:
Looks good!